projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts

```diff
@@ -71,6 +71,83 @@ export class FhirDataService {
       return teardownLogic;
     });
   }
+  addPatientData(reportBPValue: any) {
+    const resource = {
+      "resourceType": "Observation",
+      "status": "final",
+      "category": [
+        {
+          "coding": [
+            {
+              "system": "http://terminology.hl7.org/CodeSystem/observation-category",
+              "code": "vital-signs",
+              "display": "vital-signs"
+            }
+          ]
+        }
+      ],
+      "code": {
+        "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "85354-9",
+            "display": "Blood Pressure"
+          }
+        ],
+        "text": "Blood Pressure"
+      },
+      "subject": {
+        "reference": `Patient/${this.client?.patient.id}`
+      },
+      "encounter": {
+        "reference": "Encounter/23249"
+      },
```

Comment:
We should not use a hard-coded encounter ID

Reply:
Okay, I will get the encounter ID from client.encounter.